cardano-api/src/Cardano/Api/Tx.hs

```diff
@@ -409,52 +410,55 @@ getTxBody (ByronTx Byron.ATxAux { Byron.aTaTx = txbody }) =
-
-    getAlonzoTxBody :: forall ledgerera.
-                       ShelleyLedgerEra era ~ ledgerera
-                    => Ledger.Witnesses ledgerera ~ Alonzo.TxWitness ledgerera
-                    => Shelley.ShelleyBased ledgerera
-                    => ScriptDataSupportedInEra era
-                    -> Ledger.Tx ledgerera
-                    -> TxBody era
-    getAlonzoTxBody scriptDataInEra
-                    Shelley.Tx {
-                      Shelley.body = txbody,
-                      Shelley.wits = Alonzo.TxWitness'
-                                     _addrWits
-                                     _bootWits
-                                     txscripts
-                                     txdats
-                                     redeemers,
-                      Shelley.auxiliaryData = auxiliaryData
-                    } =
-      ShelleyTxBody era txbody
-                    (Map.elems txscripts)
-                    (TxBodyScriptData scriptDataInEra txdats redeemers)
-                    (strictMaybeToMaybe auxiliaryData)
+ where
+  getShelleyTxBody
+    :: forall era' ledgerera. ShelleyLedgerEra era' ~ ledgerera
+    => Ledger.Era ledgerera
+    => FromCBOR (CBOR.Annotator (Ledger.Script ledgerera))
+    => ToCBOR (Ledger.Script ledgerera)
+    => Ledger.Witnesses ledgerera ~ Shelley.WitnessSetHKD Identity ledgerera
+    => ToCBOR (Ledger.AuxiliaryData ledgerera)
+    => ToCBOR (Ledger.TxBody ledgerera)
+    => ToCBOR (Ledger.Witnesses ledgerera)
+    => ShelleyBasedEra era'
+    -> Shelley.Tx ledgerera
+    -> TxBody era'
+  getShelleyTxBody era' Shelley.Tx {
+                     Shelley.body       = txbody,
+                     Shelley.auxiliaryData = txAuxiliaryData,
+                     Shelley.wits = Shelley.WitnessSet
+                                            _addrWits
+                                             msigWits
+                                            _bootWits
+                   } =
+    ShelleyTxBody era' txbody
+                  (Map.elems msigWits)
+                  TxBodyNoScriptData
+                  (strictMaybeToMaybe txAuxiliaryData)
+
+  getAlonzoTxBody
+    :: ScriptDataSupportedInEra era
+    -> Alonzo.ValidatedTx (ShelleyLedgerEra era)
+    -> TxBody era
+  getAlonzoTxBody scriptDataInEra
+                  Alonzo.ValidatedTx {
+                    Alonzo.body = txbody,
+                    Alonzo.wits = Alonzo.TxWitness'
+                                   _addrWits
+                                   _bootWits
+                                   txscripts
+                                   txdats
+                                   redeemers,
+                    Alonzo.auxiliaryData = auxiliaryData
+                  } =
+    ShelleyTxBody era txbody
+                  (Map.elems txscripts)
+                  (TxBodyScriptData scriptDataInEra txdats redeemers)
+                  (strictMaybeToMaybe auxiliaryData)
```

Comment:
There's a lot of changes here, both specifically here and in the rest of the module. It's unclear what are the forced changes due to the update and what is reformatting. Could we try to split those up? i.e. keep _just_ the necessary changes, and if we want to do reformatting or other refactoring, have those in separate commits, either before or after the dep update.